atomics/src/Lock_Array_SYCL.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+
+// FIXME_SYCL Use SYCL_EXT_ONEAPI_DEVICE_GLOBAL when available instead
+#ifdef DESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
+
+#include <cinttypes>
+#include <desul/atomics/Lock_Array_SYCL.hpp>
+#include <sstream>
+#include <string>
+
+namespace desul {
+namespace Impl {
+SYCL_EXTERNAL
+sycl::ext::oneapi::experimental::device_global<
+    int32_t*,
+    decltype(sycl::ext::oneapi::experimental::properties(
+        sycl::ext::oneapi::experimental::device_image_scope))>
```

Comment:
What is that `device_image_scope ` property and why do we need it?

Reply:
OK I saw the comment in `Lock_Array_SYCL.hpp`.  Maybe we should define some alias along these lines
```C++
template <class T>
using sycl_device_global = cl::ext::oneapi::experimental::device_global<
     T,
     decltype(sycl::ext::oneapi::experimental::properties(
         sycl::ext::oneapi::experimental::device_image_scope))>